File: backend/DTOs/MemberQuestDto.cs
namespace GoalboundFamily.Api.DTOs;

public class MemberQuestDto
{
    public Guid HouseholdMemberId { get; set; }
    public Guid QuestId { get; set; }

    // MemberQuest fields
    public string Status { get; set; } = string.Empty;
    public int Progress { get; set; }
    public DateTime AssignedAt { get; set; }
    public DateTime? StartTime { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? ClaimedAt { get; set; }

    // Quest fields
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int XpReward { get; set; }
    public string Category { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Difficulty { get; set; } = string.Empty;
    public int Target { get; set; }
}

File: frontend/src/types/MemberQuestDto.ts
import type { QuestCategory } from './QuestCategory';

export interface MemberQuestDto {
    householdMemberId: string; // Guid -> string
    questId: string; // Guid -> string
  
    // MemberQuest fields
    status: string;
    progress: number;
    assignedAt: string; // DateTime -> string (ISO format)
    startTime?: string; // nullable DateTime
    completedAt?: string; // nullable DateTime
    claimedAt?: string; // nullable DateTime
  
    // Quest fields
    title: string;
    description: string;
    xpReward: number;
    category: QuestCategory;
    type: string;
    difficulty: string;
    target: number;

    timeLimitSeconds?: number | null;
  }

